### DISCUSSION TITLE
Using ClusterData to split a graph into subgraphs

### DISCUSSION THREAD

**Discussion by yangyangdotcom** [pyg-team/pytorch_geometric#6602] [Q&A] [answered]
I am trying to use ClustreData to partiton my graphs into 8 subgraphs. However for some of the graph it would return an index error as such:

`IndexError: index 236 is out of bounds for dimension 0 with size 235`

Any help would be appreciated.

**Comment by wsad1** [comment, marked as answer]
Hi @yangyangdotcom could you please share the following to help us debug this.
1. Run `data.validate` and see what happens.
2.  Post the full stack trace. it will help us understand the issue better.
3. share the the number of node and edges in you graph and the inputs you passed to `ClusterData`.

**Reply by yangyangdotcom** [reply]
Thanks for trplying. Here it is:

1.
`<bound method Data.validate of Data(x=[219, 20], edge_index=[2, 2422], edge_attr=[2422], seq='LDKQQVQLLAEMCILIDENDNKIGAETKKNCHLNENIEKGLLHRAFSVFLFNTENKLLLQQRSDAKITFPGCFTNTCCSHPLSNPAELEESDALGVRRAAQRRLKAELGIPLEEVPPEEINYLTRIHYKAQSDGIWGEHEIDYILLVRKNVTLNPDPNEIKSYCYVSKEELKELLKKAASGEIKITPWFKIIAATFLFKWWDNLNHLNQFVDHEKIYRM', name=/Users/usr/Documents/Classification/Graph/Data/raw/2ICJ.pdb, adj=[219, 219])>`

2.
Traceback (most recent call last):
  File "/Users/usr/Documents/Classification/Graph-subgraph/testing.py", line 63, in <module>
    partiton_graph()
  File "/Users/usr/Documents/Classification/Graph-subgraph/testing.py", line 22, in partiton_graph
    cluster = ClusterData(data[0],4,save_dir = "/Users/usr/Documents/Classification/Graph-subgraph/subgraph_data/")
  File "/Users/usr/opt/anaconda3/envs/PPI_GNN_seqvec/lib/python3.9/site-packages/torch_geometric/loader/cluster.py", line 58, in __init__
    self.data = self.__permute_data__(data, perm, adj)
  File "/Users/usr/opt/anaconda3/envs/PPI_GNN_seqvec/lib/python3.9/site-packages/torch_geometric/loader/cluster.py", line 66, in __permute_data__
    out[key] = value[node_idx]
IndexError: index 219 is out of bounds for dimension 0 with size 219


3.
My graph has 219 nodes and 1211 edges. I pass in the `data` from the first bullet point to the `ClusterData`

**Reply by wsad1** [reply]
Sorry i meant call `data.validate()`. From the error it looks like your `edge_index` might have values greater thank 218, but the nodes are indexed from 0-218, can you confirm this.

**Reply by yangyangdotcom** [reply]
It returns True when I call `data.validate()`. That is correct, my graph has more edges than the nodes. I try to partition other dataset that has more edges than nodes and it works. Not sure why it doesn't work on my dataset.

**Reply by wsad1** [reply]
Its ok for your graph to have more edges than nodes. But the edge should be between nodes that exist, so if your graph has 219 nodes then `edge_index` has to have entries between [0,218].

**Reply by yangyangdotcom** [reply]
I think I found the problem. This happen when a path is given as the `save_dir`. Not sure why, but the error disappear after I remove the `save_dir`.
